spec/controllers/tags_controller_spec.rb

```diff
@@ -173,6 +173,45 @@
     end
   end
 
+  describe "show_hidden" do
+    let!(:fandom1) { create(:fandom, canonical: true) }
+    let!(:freeform1) { create(:freeform, canonical: false) }
+    let!(:character1) { create(:character, canonical: false) }
+    let!(:character2) { create(:character, canonical: false) }
+    let!(:work) do
```

Comment:
You can remove the 2 ajax success tests, which we already cover in Cucumber.

For the remaining 2 tests, we're just looking for the error message "you need to have JavaScript". We don't actually need to set up the tags, so the test setup can just be:

```rb
describe "show_hidden" do
  let(:work) { create(:work, posted: true) }
```

It's fine to use `let` over `let!` here and have the work be lazily created when `work.id` is first used.

Reply:
Sure - resolved in 20ee3b511.